app/src/main/res/layout/activity_main.xml

```diff
@@ -13,6 +13,22 @@
     tools:context=".MainActivity"
     android:orientation="vertical">
 
+    <TextView
```

Comment:
You have to implement a custom view. Don't use predefined tags like textView or anything define your own View class BatteryView or something.